### PR TITLE
Document CI requirements

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: MAINTAINING.md [hunk 1]
```diff
@@ -33,6 +36,14 @@
     * Documents in the `docs` directory are consistent with new changes
     * All combinators have time and space complexity annotations
     * All combinators have `since` annotation
+    * Unreleased combinators should be marked with `pre-release` or
+      `internal` annotations
```
Comment (adithyaov) [inline MAINTAINING.md:40]:
```suggestion
    * Unreleased combinators should be marked with `Pre-release` or
      `Internal` annotations
```

Comment (adithyaov) [reply MAINTAINING.md:40]: All small also works.

File: MAINTAINING.md [hunk 2]
```diff
@@ -33,6 +36,14 @@
...
+
+* _Build and Test_:
+
+    * All CIs are green
+    * Manually build and test for all flags (esp. `dev` flag) not covered by CIs
```
Comment (adithyaov) [inline MAINTAINING.md:45]: We should probably list the flags here (or somewhere else) for convinence.

**Review by adithyaov — APPROVED** [review]

File: MAINTAINING.md (hunk 2 above)
Comment (harendra-kumar) [reply MAINTAINING.md:45]: They are all listed on the hackage index page and in the cabal file.

**Pull request merged**